src/main/java/sg/ncl/MainController.java

```diff
@@ -1173,6 +1130,66 @@ public String editAccountDetails(
         return "redirect:/account_settings";
     }
 
+    private void checkUserUpdate(@ModelAttribute("editUser") User2 editUser, RedirectAttributes redirectAttributes, User2 originalUser) {
+        if (!originalUser.getFirstName().equals(editUser.getFirstName())) {
+            redirectAttributes.addFlashAttribute("editFirstName", SUCCESS);
+        }
+        if (!originalUser.getLastName().equals(editUser.getLastName())) {
+            redirectAttributes.addFlashAttribute("editLastName", SUCCESS);
+        }
+        if (!originalUser.getPhone().equals(editUser.getPhone())) {
+            redirectAttributes.addFlashAttribute("editPhone", SUCCESS);
+        }
+        if (!originalUser.getJobTitle().equals(editUser.getJobTitle())) {
+            redirectAttributes.addFlashAttribute("editJobTitle", SUCCESS);
+        }
+        if (!originalUser.getInstitution().equals(editUser.getInstitution())) {
+            redirectAttributes.addFlashAttribute("editInstitution", SUCCESS);
+        }
+        if (!originalUser.getCountry().equals(editUser.getCountry())) {
+            redirectAttributes.addFlashAttribute("editCountry", SUCCESS);
+        }
+    }
+
+    private boolean checkEditUserFields(@ModelAttribute("editUser") User2 editUser, RedirectAttributes redirectAttributes, String editPhrase) {
+        boolean errorsFound = false;
+        if (!errorsFound && editUser.getFirstName().isEmpty()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)